aws_lambda_powertools/utilities/parameters/ssm.py

```diff
@@ -188,7 +188,7 @@ def _get(self, name: str, decrypt: bool = False, **sdk_options) -> str:
 
         return self.client.get_parameter(**sdk_options)["Parameter"]["Value"]
 
-    def _get_multiple(self, path: str, decrypt: bool = False, recursive: bool = False, **sdk_options) -> Dict[str, str]:
+    def _get_multiple(self, path: str, decrypt: Optional[bool] = None, recursive: bool = False, **sdk_options) -> Dict[str, str]:
```

Comment:
I'm not sure we would change this method. I think that due to our error, we did not implement `get_multiple` in the `SSMProvider` class. We already implemented `get` and we should do the same for `get_multiple`. 

Makes more sense for us to put the `get_multiple` method in this class and do the appropriate checks for variables/env and others in it and then call the superclass.

It would be something more or less like this:

```python
def get_multiple( # type: ignore[override]
        self,
        path: str,
        max_age: Optional[int] = None,
        transform: TransformOptions = None,
        raise_on_transform_error: bool = False,
        decrypt: Optional[bool] = None,
        force_fetch: bool = False,
        **sdk_options,
    ) -> Union[Dict[str, str], Dict[str, dict], Dict[str, bytes]]:
        """
        Retrieve multiple parameters based on a path prefix

        Parameters
        ----------
        path: str
            Parameter path used to retrieve multiple parameters
        max_age: int, optional
            Maximum age of the cached value
        transform: str, optional
            Optional transformation of the parameter value. Supported values
            are "json" for JSON strings, "binary" for base 64 encoded
            values or "auto" which looks at the attribute key to determine the type.
        raise_on_transform_error: bool, optional
            Raises an exception if any transform fails, otherwise this will
            return a None value for each transform that failed
        force_fetch: bool, optional
            Force update even before a cached item has expired, defaults to False
        sdk_options: dict, optional
            Arguments that will be passed directly to the underlying API call

        Raises
        ------
        GetParameterError
            When the parameter provider fails to retrieve parameter values for
            a given path.
        TransformParameterError
            When the parameter provider fails to transform a parameter value.
        """
        # If max_age is not set, resolve it from the environment variable, defaulting to DEFAULT_MAX_AGE_SECS
        max_age = resolve_max_age(env=os.getenv(constants.PARAMETERS_MAX_AGE_ENV, DEFAULT_MAX_AGE_SECS), choice=max_age)

        # If decrypt is not set, resolve it from the environment variable, defaulting to False
        decrypt = resolve_truthy_env_var_choice(
            env=os.getenv(constants.PARAMETERS_SSM_DECRYPT_ENV, "false"),
            choice=decrypt,
        )

        # Add to `decrypt` sdk_options to we can have an explicit option for this
        sdk_options["decrypt"] = decrypt

        return super().get_multiple(path, max_age, transform, force_fetch, raise_on_transform_error, **sdk_options)
```

Make sense for you?

Reply:
Yes that makes sense, seems more in-line with the existing pattern for `get`.

Would you like me to make the change, or will you do it?

Reply:
`decrypt` is missing from the doc, btw

Reply:
> Yes that makes sense, seems more in-line with the existing pattern for `get`.
> 
> Would you like me to make the change, or will you do it?

If you can do it, please do. :1st_place_medal:

Reply:
> `decrypt` is missing from the doc, btw

Hmm where?

Reply:
In

```
Parameters
        ----------
        path: str
            Parameter path used to retrieve multiple parameters
        max_age: int, optional
            Maximum age of the cached value
        transform: str, optional
            Optional transformation of the parameter value. Supported values
            are "json" for JSON strings, "binary" for base 64 encoded
            values or "auto" which looks at the attribute key to determine the type.
        raise_on_transform_error: bool, optional
            Raises an exception if any transform fails, otherwise this will
            return a None value for each transform that failed
        force_fetch: bool, optional
            Force update even before a cached item has expired, defaults to False
        sdk_options: dict, optional
            Arguments that will be passed directly to the underlying API call
```

but as I am making the change, I will add it.

Reply:
In `get` we have `decrypt` before `force_fetch`, which breaks liskov substitution for people using the arguments positionally and requiring the base class.

The change that you've suggested above does likewise. I can fix this by putting `decrypt` at the end, before `sdk_options`. Should I do this, or keep it consistent with `get` (breaking liskov substitution)?

Reply:
I think we should keep consistence with `get` because people can easily switch between `ssm_provider.get("/leandro/a", decrypt=True)` and `ssm_provider.get_multipe("/leandro/", decrypt=True)`.

please add a note in the comment to review this in Powertools V3, ok?

Reply:
As the method in `base` is not currently overridden, this will be a breaking change for those consumers using the method arguments positionally. Should I proceed?

Reply:
@leandrodamascena I'm a little confused now. Is the proposal to create a `get_multiple` which overrides the `base`, but read from `sdk_options['decrypt']`, or, to keep the implementation as it already is in the PR?

Reply:
Got it. 👍 

Should I update any comments?

Reply:
As far as other considerations, I think it's fine. The `_get_multiple` here is only called in `base`, which is passing **kwargs (and has no understanding of the shape of `decrypt`), so I don't think the minor change in signature is important.